src/main/java/org/jenkinsci/plugins/blueoceandisplayurl/BlueOceanDisplayURLImpl.java

```diff
@@ -69,12 +69,12 @@ public String getRunURL(Run<?, ?> run) {
             WorkflowJob job = ((WorkflowRun) run).getParent();
             if (job.getParent() instanceof MultiBranchProject) {
                 String jobURL = getJobURL(organization, ((MultiBranchProject) job.getParent()));
-                return String.format("%sdetail/%s/%d/", jobURL, Util.rawEncode(job.getDisplayName()), run.getNumber());
+                return String.format("%sdetail/%s/%d/", jobURL, job.getName(), run.getNumber());
             }
```

Comment:
@imeredith How does frontend use this name? If it uses it to call API, then with custom display name, it would never have worked as we don't have API to look for jobs using displayName.

In any case, if API expects double encoded name, then its better to encode name (even if it results in double encoding). you probably want to test it with front end.

Reply:
Most jobs dont use display name. And This problem was  with branches. Double encoding breaks the routing

Reply:
Its ok as long as we know it doesn't break UI.